InventoryCatalog/etc/di.xml

```diff
@@ -162,4 +161,13 @@
         <plugin name="inventoryUpdate" type="Magento\InventoryCatalog\Plugin\Catalog\Controller\Adminhtml\Product\Action\Attribute\Save\ProcessInventoryPlugin"/>
         <plugin name="massAction" disabled="true"/>
     </type>
+    <type name="Magento\Catalog\Model\Product\Type\Simple">
+        <plugin name="is_simple_product_salable" type="Magento\InventoryCatalog\Plugin\Catalog\Model\Type\Simple\IsSalablePlugin"/>
+    </type>
```

Comment:
this part of improvement we agreed to postpone till later. 
this is not part of async salability indexer, but premature assignment of salability done by Catalog